src/transformers/models/electra/modeling_flax_electra.py

```diff
@@ -0,0 +1,689 @@
+        if token_type_ids is None:
+            token_type_ids = jnp.ones_like(input_ids)
+
+        if position_ids is None:
+            position_ids = jnp.broadcast_to(jnp.arange(jnp.atleast_2d(input_ids).shape[-1]), input_ids.shape)
+
+        if attention_mask is None:
+            attention_mask = jnp.ones_like(input_ids)
+
+        # Handle any PRNG if needed
+        rngs = {}
+        if dropout_rng is not None:
+            rngs["dropout"] = dropout_rng
+
+        return self.module.apply(
+            {"params": params or self.params},
+            jnp.array(input_ids, dtype="i4"),
+            jnp.array(attention_mask, dtype="i4"),
+            jnp.array(token_type_ids, dtype="i4"),
+            jnp.array(position_ids, dtype="i4"),
+            not train,
+            rngs=rngs,
+        )
+
+
+class FlaxElectraModule(nn.Module):
+    config: ElectraConfig
+    dtype: jnp.dtype = jnp.float32  # the dtype of the computation
+
+    def setup(self):
+        self.embeddings = FlaxElectraEmbeddings(self.config, dtype=self.dtype)
+        if self.config.embedding_size != self.config.hidden_size:
+            self.embeddings_project = nn.Dense(self.config.hidden_size)
+        self.encoder = FlaxElectraEncoder(self.config, dtype=self.dtype)
+
+    def __call__(self, input_ids, attention_mask, token_type_ids, position_ids, deterministic: bool = True):
+        embeddings = self.embeddings(
+            input_ids, token_type_ids, position_ids, attention_mask, deterministic=deterministic
+        )
+        if hasattr(self, "embeddings_project"):
+            embeddings = self.embeddings_project(embeddings)
+
+        hidden_states = self.encoder(embeddings, attention_mask, deterministic=deterministic)
+        return (hidden_states,)
+
+
+@add_start_docstrings(
+    "The bare Electra Model transformer outputting raw hidden-states without any specific head on top.",
+    ELECTRA_START_DOCSTRING,
+)
+class FlaxElectraModel(FlaxElectraPreTrainedModel):
+    module_class = FlaxElectraModule
+
+
+class FlaxElectraForMaskedLMModule(nn.Module):
+    config: ElectraConfig
+    dtype: jnp.dtype = jnp.float32
+
+    def setup(self):
+        self.electra = FlaxElectraModule(config=self.config, dtype=self.dtype)
+        self.generator_predictions = FlaxElectraGeneratorPredictions(config=self.config)
+        if self.config.tie_word_embeddings:
+            self.generator_lm_head = TiedDense(self.config.vocab_size, dtype=self.dtype)
+        else:
+            self.generator_lm_head = nn.Dense(self.config.vocab_size, dtype=self.dtype)
+
+    def __call__(
+        self, input_ids, attention_mask=None, token_type_ids=None, position_ids=None, deterministic: bool = True
+    ):
+        hidden_states = self.electra(
+            input_ids, attention_mask, token_type_ids, position_ids, deterministic=deterministic
+        )[0]
+        prediction_scores = self.generator_predictions(hidden_states)
+
+        if self.config.tie_word_embeddings:
```

Comment:
Actually, Electra doesn't tie the word embeddings at all, so I don't think we need the logic here :-)

To give a bit more detail here:

In PyTorch, we call the function [_tie_or_clone_weights()](https://github.com/huggingface/transformers/blob/04ab2ca639ee6fd1002ce0a498d892245f0c9093/src/transformers/modeling_utils.py#L581) on every model class that:
- has `self.config.tie_word_embeddings` set to True (this is the case for ELECTRA)
- **and** that overwrites the function `get_output_embeddings()` to return output embeddings != None (this is **not** the case for ELECTRA)

=> no weights are tied

Reply:
Hmm this is in fact weird. I originally didn't include the tied embeddings but I wasn't passing the test without it. I'll try again removing this

Reply:
So I just double checked and it seems that weight tying does happen in electra (see [here](https://github.com/huggingface/transformers/blob/bc2571e61c985ec82819cf01ad038342771c94d0/src/transformers/models/electra/modeling_electra.py#L1087)) and I cannot pass the tests without. However, following @sgugger advice, I moved the `TiedDense` inside the flax_electra_module (if you prefered to keep it somewhere reusable let me know)

Reply:
You are totally right - thanks for double checking! I missed the `get_output_embeddings()` function